Python/ds/SinglyLinkedList.py

```diff
@@ -0,0 +1,144 @@
+        self.head = nnode
+
+    def insertNodeAtEnd(self, ndata):
+        nnode = Node(ndata)
+        if self.head == None:
+            self.head = nnode
+        else:
+            last = self.head
+            while last.next != None:
+                last = last.next
+            last.next = nnode
+    
+    def insertNodeAtPosition(self, index, ndata):
+        if index == 1:
+            insertNodeAtFront(self, ndata)
+        else:
+            cur = 1
+            pos = self.head
+            while cur < index-1 and pos != None:
+                pos = pos.next
+                cur += 1
+            if pos == None:
+                insertNodeAtEnd(self, ndata)
+            else:
+                nnode = Node(ndata)
+                nnode.next = pos.next
+                pos.next = nnode
+
+    def deleteNodeAtFront(self):
+        if self.head == None:
+            print("Empty list, nothing to delete!")
+        else:
+            self.head = self.head.next
+
+    def deleteNodeAtEnd(self):
+        if self.head == None:
+            print("Empty list, nothing to delete!")
+        elif self.head.next == None:
+            self.head=None
+        else:
+            pos = self.head
+            while pos.next.next != None:
+                pos = pos.next
+            pos.next = None
+
+    def deleteNodeAtPosition(self, index):
+        if self.head == None:
+            print("Empty list, nothing to delete!")
+            return
+        pos = self.head
+        if index == 1:
+            self.head = pos.next
+            temp = None
+            return
+        for i in range(1,index-1):
+            pos = pos.next
+            if pos == None:
+                break
+        if pos == None or pos.next == None:
+            print("Index more than number of nodes!")
+        else:
+            next = pos.next.next
+            pos.next = None
+            pos.next = next
+
+    def traverseList(self):
+        if self.head == None:
+            print("Empty list, nothing to print!")
+        else:
+            n = self.head
+            while n != None:
+                print(n.data)
+                n = n.next
+
+    def countNodes(self):
+        cnt = 0
+        ind = self.head
+        while ind != None:
+            cnt += 1
+            ind = ind.next
+        print(cnt)
+
+if __name__=='__main__':
+    newLinkedList = LinkedList()
+    print("SINGLY LINKED LIST")
+    print("1. Insert Node at Front")
+    print("2. Insert Node at End")
+    print("3. Insert Node at Position")
+    print("4. Delete Node at Front")
+    print("5. Delete Node at End")
+    print("6. Delete Node at Position")
+    print("7. Print the List")
+    print("8. Count nodes in the List")
+    print("9. Exit")
+    print("--Use 1-based indexing--")
+    while(True):
+        print("Enter your choice: ")
+        ch = int(input())
+        if ch == 1:
+            print("Enter the element to insert: ")
+            ndata = int(input())
```

Comment:
Pass output statements in `input()`
Like:
`ndata = int(input("Enter the element to insert: "))`

Reply:
I have changed it to 'ind'